src/components/connect.js

```diff
@@ -19,6 +19,14 @@ function getDisplayName(WrappedComponent) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component'
 }
 
+function getMapDispatch(value) {
+  switch (true) {
+    case (typeof value === 'function'): return value
```

Comment:
I agree that a `switch` statement here can be unusual. But we can translate that directly to guard statements:
```javascript
function getMapDispatch(value) {
  if (typeof value === 'function') {
    return value
  }

  if (isPlainObject(value)) {
    return wrapActionCreators(value)
  }
    
  return defaultMapDispatchToProps
}
```